src/mq.c

```diff
@@ -418,32 +534,34 @@ amqp_bytes_t mq_declare_new_queue(const char *name)
  *
  * Returns: 0 if successful and -1 otherwise.
  */
-int mq_bind_queue(amqp_bytes_t queue,
-			      const char *exchange,
-			      const char *routing_key)
+int mq_prepare_direct_queue(amqp_bytes_t queue, const char *exchange,
+			    const char *routing_key)
 {
-	if (exchange == NULL || routing_key == NULL)
-		return -1;
-
-	/* Declare the exchange as durable */
-	amqp_exchange_declare(mq_ctx.conn, 1,
-			amqp_cstring_bytes(exchange),
-			amqp_cstring_bytes("topic"),
-			0 /* passive*/,
-			1 /* durable */,
-			0 /* auto_delete*/,
-			0 /* internal */,
-			amqp_empty_table);
+	return mq_prepare_queue(queue, exchange, "direct", routing_key);
+}
 
-	/* Set up to bind a queue to an exchange */
-	amqp_queue_bind(mq_ctx.conn, 1, queue,
-			amqp_cstring_bytes(exchange),
-			amqp_cstring_bytes(routing_key),
+/**
+ * mq_consumer_queue:
+ * @queue: queue that is going to be consume
+ *
+ * Start a queue consumer.
+ *
+ * Returns: 0 if successful and -1 otherwise.
+ */
+int mq_consumer_queue(amqp_bytes_t queue)
+{
+	/* Start a queue consumer */
+	amqp_basic_consume(mq_ctx.conn, 1,
```

Comment:
Add a comment informing whats is 1 or why it should be 1.

Reply:
I think that this is not related to this PR

Reply:
I do not think there is a problem to modify it on this PR since you are refactoring this part of code but it makes sense to me.